common/djangoapps/third_party_auth/saml.py

```diff
@@ -41,10 +41,18 @@ def auth_url(self):
         if not self._config.enabled:
             log.error('SAML authentication is not enabled')
             raise Http404
+
         # TODO: remove this check once the fix is merged upstream:
         # https://github.com/omab/python-social-auth/pull/821
         if 'idp' not in self.strategy.request_data():
             raise AuthMissingParameter(self, 'idp')
+
+        from .models import SAMLProviderConfig
```

Comment:
Yup, that's what I meant. Neither the relative nor top-level imports worked when the module was first loaded, so I stuck with doing a local import of `SAMLProviderConfig`, just like [`get_idp()`](https://github.com/edx/edx-platform/pull/13474/files#diff-31c7777edd76b5f6c6190122337c991cL22) does.